connect/runtime/src/test/java/org/apache/kafka/connect/runtime/rest/resources/ConnectorsResourceTest.java

```diff
@@ -899,250 +750,199 @@ public void testFenceZombiesWithInternalRequestSignature() throws Throwable {
-        PowerMock.replay(workerConfig);
+        when(workerConfig.getBoolean(TOPIC_TRACKING_ENABLE_CONFIG)).thenReturn(true);
+        when(workerConfig.getBoolean(TOPIC_TRACKING_ALLOW_RESET_CONFIG)).thenReturn(false);
+        HttpHeaders headers = mock(HttpHeaders.class);
         connectorsResource = new ConnectorsResource(herder, workerConfig);
-        PowerMock.replayAll();
 
         Exception e = assertThrows(ConnectRestException.class,
             () -> connectorsResource.resetConnectorActiveTopics(CONNECTOR_NAME, headers));
         assertEquals("Topic tracking reset is disabled.", e.getMessage());
-        PowerMock.verifyAll();
     }
 
     @Test
     public void testConnectorActiveTopics() {
-        PowerMock.reset(workerConfig);
-        EasyMock.expect(workerConfig.getBoolean(TOPIC_TRACKING_ENABLE_CONFIG)).andReturn(true);
-        EasyMock.expect(workerConfig.getBoolean(TOPIC_TRACKING_ALLOW_RESET_CONFIG)).andReturn(true);
-        EasyMock.expect(herder.connectorActiveTopics(CONNECTOR_NAME))
-                .andReturn(new ActiveTopicsInfo(CONNECTOR_NAME, CONNECTOR_ACTIVE_TOPICS));
-        PowerMock.replay(workerConfig);
+        when(workerConfig.getBoolean(TOPIC_TRACKING_ENABLE_CONFIG)).thenReturn(true);
+        when(workerConfig.getBoolean(TOPIC_TRACKING_ALLOW_RESET_CONFIG)).thenReturn(true);
+        when(herder.connectorActiveTopics(CONNECTOR_NAME))
+            .thenReturn(new ActiveTopicsInfo(CONNECTOR_NAME, CONNECTOR_ACTIVE_TOPICS));
         connectorsResource = new ConnectorsResource(herder, workerConfig);
-        PowerMock.replayAll();
 
         Response response = connectorsResource.getConnectorActiveTopics(CONNECTOR_NAME);
         assertEquals(Response.Status.OK.getStatusCode(), response.getStatus());
         Map<String, Map<String, Object>> body = (Map<String, Map<String, Object>>) response.getEntity();
         assertEquals(CONNECTOR_NAME, ((ActiveTopicsInfo) body.get(CONNECTOR_NAME)).connector());
         assertEquals(new HashSet<>(CONNECTOR_ACTIVE_TOPICS),
                 ((ActiveTopicsInfo) body.get(CONNECTOR_NAME)).topics());
-        PowerMock.verifyAll();
     }
 
     @Test
     public void testResetConnectorActiveTopics() {
-        PowerMock.reset(workerConfig);
-        EasyMock.expect(workerConfig.getBoolean(TOPIC_TRACKING_ENABLE_CONFIG)).andReturn(true);
-        EasyMock.expect(workerConfig.getBoolean(TOPIC_TRACKING_ALLOW_RESET_CONFIG)).andReturn(true);
-        HttpHeaders headers = EasyMock.mock(HttpHeaders.class);
+        when(workerConfig.getBoolean(TOPIC_TRACKING_ENABLE_CONFIG)).thenReturn(true);
+        when(workerConfig.getBoolean(TOPIC_TRACKING_ALLOW_RESET_CONFIG)).thenReturn(true);
+        HttpHeaders headers = mock(HttpHeaders.class);
         herder.resetConnectorActiveTopics(CONNECTOR_NAME);
-        EasyMock.expectLastCall();
-        PowerMock.replay(workerConfig);
+        verify(herder).resetConnectorActiveTopics(CONNECTOR_NAME);
```

Comment:
This part should be moved after the call to `connectorsResource::resetConnectorActiveTopics`.

Reply:
This change ends up failing the test with

```
herder.resetConnectorActiveTopics("test");
Wanted 1 time:
-> at org.apache.kafka.connect.runtime.rest.resources.ConnectorsResourceTest.testResetConnectorActiveTopics(ConnectorsResourceTest.java:912)
But was 2 times:
-> at org.apache.kafka.connect.runtime.rest.resources.ConnectorsResourceTest.testResetConnectorActiveTopics(ConnectorsResourceTest.java:908)
-> at org.apache.kafka.connect.runtime.rest.resources.ConnectorsResource.resetConnectorActiveTopics(ConnectorsResource.java:234)
```

Should I change `verify(herder).resetConnectorActiveTopics(CONNECTOR_NAME);` to use `atLeastOnce(), i.e.

```java
verify(herder, atLeastOnce).resetConnectorActiveTopics(CONNECTOR_NAME);
```

Reply:
This was fixed by removing the direct call to `herder::resetConnectorActiveTopics` from the test case.